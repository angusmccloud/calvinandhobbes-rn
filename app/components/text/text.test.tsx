import React from 'react';
import renderer from 'react-test-renderer';
import Text from './text';

describe('Text Component rendered correctly', () => {
  it('XXS Text Size', () => {
    const text = renderer.create(<Text size="XXS">XXS Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('XS Text Size', () => {
    const text = renderer.create(<Text size="XS">XS Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('S Text Size', () => {
    const text = renderer.create(<Text size="S">S Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('M Text Size', () => {
    const text = renderer.create(<Text size="M">M Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('L Text Size', () => {
    const text = renderer.create(<Text size="L">L Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('XL Text Size', () => {
    const text = renderer.create(<Text size="XL">XL Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('XXL Text Size', () => {
    const text = renderer.create(<Text size="XXL">XXL Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('No Size Defined', () => {
    const text = renderer.create(<Text>No Size Defined Text</Text>);
    expect(text).toMatchSnapshot();
  });
  it('Bold Text', () => {
    const text = renderer.create(<Text bold>Bold Text</Text>);
    expect(text).toMatchSnapshot();
  });
});
