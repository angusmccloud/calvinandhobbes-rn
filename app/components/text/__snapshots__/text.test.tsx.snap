// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component rendered correctly Bold Text 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir-Heavy",
      "fontSize": 36,
      "fontWeight": "700",
    }
  }
>
  Bold Text
</Text>
`;

exports[`Text Component rendered correctly L Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 44,
      "fontWeight": "400",
    }
  }
>
  L Text
</Text>
`;

exports[`Text Component rendered correctly M Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 36,
      "fontWeight": "400",
    }
  }
>
  M Text
</Text>
`;

exports[`Text Component rendered correctly No Size Defined 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 36,
      "fontWeight": "400",
    }
  }
>
  No Size Defined Text
</Text>
`;

exports[`Text Component rendered correctly S Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 32,
      "fontWeight": "400",
    }
  }
>
  S Text
</Text>
`;

exports[`Text Component rendered correctly XL Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 52,
      "fontWeight": "400",
    }
  }
>
  XL Text
</Text>
`;

exports[`Text Component rendered correctly XS Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 28,
      "fontWeight": "400",
    }
  }
>
  XS Text
</Text>
`;

exports[`Text Component rendered correctly XXL Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 72,
      "fontWeight": "400",
    }
  }
>
  XXL Text
</Text>
`;

exports[`Text Component rendered correctly XXS Text Size 1`] = `
<Text
  allowFontScaling={false}
  numberOfLines={0}
  style={
    Object {
      "color": "#323030",
      "fontFamily": "Avenir",
      "fontSize": 24,
      "fontWeight": "400",
    }
  }
>
  XXS Text
</Text>
`;
